EventFilter/TotemRawToDigi/src/RawToDigiConverter.cc

```diff
@@ -0,0 +1,271 @@
+      }
+    }
+
+    // if there were errors, put the information to ees buffer
+    if (verbosity > 0 && !fes.rdbuf()->str().empty())
+    {
+      string message = (stopProcessing) ? "(and will be dropped)" : "(but will be used though)";
+      if (verbosity > 2)
+      {
+        ees << "  Frame at " << fr.Position() << " seems corrupted " << message << ":\n";
+        ees << fes.rdbuf();
+      } else
+        ees << "  Frame at " << fr.Position() << " seems corrupted " << message << ".\n";
+    }
+
+    // if there were serious errors, do not process this frame
+    if (stopProcessing)
+      continue;
+    
+    // fill EC and BC values to the statistics
+    if (fr.Data()->isECPresent())
+      ECChecker.Fill(fr.Data()->getEC(), fr.Position());
+
+    if (fr.Data()->isBCPresent())
+      BCChecker.Fill(fr.Data()->getBC(), fr.Position());
+  }
+
+  // analyze EC and BC statistics
+  if (testECMostFrequent != tfNoTest)
+    ECChecker.Analyze(status, (testECMostFrequent == tfErr), ees);
+
+  if (testBCMostFrequent != tfNoTest)
+    BCChecker.Analyze(status, (testBCMostFrequent == tfErr), ees);
+
+  // save the information about missing frames to conversionStatus
+  for (const auto &it : missingFrames)
+  {
+    TotemVFATStatus &actualStatus = status[it.first];
+    actualStatus.setMissing();
+    if (verbosity > 1) 
+      ees << "Frame for VFAT " << it.first << " is not present in the data.\n";
+  }
+
+  // print error message
+  if (verbosity > 0 && !ees.rdbuf()->str().empty())
+  {
+    if (verbosity > 1)
+      cerr << "event contains the following problems:\n" << ees.rdbuf() << endl;
+    else
+      cerr << "event contains problems." << endl;
+  }
+
+  // increase error counters
+  if (printErrorSummary)
+  {
+    for (const auto &it : status)
+    {
+      if (!it.second.isOK())
+      {
+        auto &m = errorSummary[it.first];
+        m[it.second]++;
+      }
+    }
+  }
+}
+
+//----------------------------------------------------------------------------------------------------
+
+void RawToDigiConverter::Run(const VFATFrameCollection &input,
+  const TotemDAQMapping &mapping, const TotemAnalysisMask &analysisMask,
+  DetSetVector<TotemRPDigi> &rpData, DetSetVector<TotemVFATStatus> &finalStatus)
+{
+  // map to link TotemFramePosition to the conversion status
+  map<TotemFramePosition, TotemVFATStatus> status;
+
+  // common processing - frame validation
+  RunCommon(input, mapping, status);
+
+  // second loop over data
+  for (VFATFrameCollection::Iterator fr(&input); !fr.IsEnd(); fr.Next())
+  {
+    auto mappingIter = mapping.VFATMapping.find(fr.Position());
+    if (mappingIter == mapping.VFATMapping.end())
+      continue;
+  
+    // check whether the data come from RP VFATs
+    if (mappingIter->second.symbolicID.subSystem != TotemSymbID::RP)
+      throw cms::Exception("RawToDigiConverter::Run") << "VFAT is not from RP. subSystem = " <<
+        mappingIter->second.symbolicID.subSystem << endl;
```

Comment:
If this can happen due to occasional/rare data corruption, the logic has to be changed to issue a LogError or LogWarning and continue with some dummy/empty output.